dalibo/temboard-agent
# Build debian package with FPM

This PR propose

- to build .deb from PyPI release instead of master
- to build .deb with public code anyone can contribute
- to test .deb workflow end-to-end with reprepro.

# Testing instructions

1. Read [packaging/README.rst](https://github.com/bersace/temboard-agent/blob/debian-fpm/packaging/README.rst)
1. `rm -rf dist/` at top level.
1. build **all** packages with a single `make build`.
1. Review logs.
1. Follow `README.rst` to run a test APT repository with `make reprepro`.
1. Upload wheezy package with `dput -c dput.cf oldstable ../dist/*_wheezy.changes`
1. Upload jessie package with `dput -c dput.cf stable ../dist/*_jessie.changes`
1. Upload stretch package with `dput -c dput.cf testing ../dist/*_stretch.changes`
1. Test each package following `README.rst`.
   1. Create a guest with the corresponding distribution (e.g. with LXC)
   1. Edit sources.list, add GPG key and update
   1. Install postgresql
   1. Install `temboard-agent`.
   1. run `temboard-agent -c /usr/share/temboard-agent/quickstart/temboard-agent.conf`
1. Ensure old `debian/` and `make_deb.sh` is not referenced elsewhere.
